File: AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/MAX/MRECs/Table View/ALMAXMRecTableViewCell.m
//
//  ALMAXMRecTableViewCell.m
//  AppLovin MAX Demo App - ObjC
//
//  Created by Alan Cao on 6/30/22.
//  Copyright © 2022 AppLovin Corporation. All rights reserved.
//

#import "ALMAXMRecTableViewCell.h"
#import <AppLovinSDK/AppLovinSDK.h>

@interface ALMAXMRecTableViewCell()

@property (nonatomic, strong, nullable) MAAdView *adView;

@end

@implementation ALMAXMRecTableViewCell

- (void)prepareForReuse
{
    [super prepareForReuse];
    
    for ( UIView *subview in self.contentView.subviews )
    {
        if ( [subview isKindOfClass: [MAAdView class]] )
        {
            [subview removeFromSuperview];
        }
    }
}

- (void)configureWith:(MAAdView *)adView
{
    self.adView = adView;
    self.adView.backgroundColor = [UIColor blackColor];
    self.adView.translatesAutoresizingMaskIntoConstraints = false;
    [self.adView startAutoRefresh];
    
    [self.contentView addSubview: self.adView];
    [NSLayoutConstraint activateConstraints: @[
        [self.adView.widthAnchor constraintEqualToConstant: 300],
        [self.adView.heightAnchor constraintEqualToConstant: 250],
        [self.adView.centerXAnchor constraintEqualToAnchor: self.contentView.centerXAnchor],
        [self.adView.centerYAnchor constraintEqualToAnchor: self.contentView.centerYAnchor],
        [self.adView.bottomAnchor constraintLessThanOrEqualToAnchor: self.contentView.bottomAnchor]
    ]];
}

- (void)stopAutoRefresh
{
    [self.adView setExtraParameterForKey: @"allow_pause_auto_refresh_immediately" value: @"true"];
    [self.adView stopAutoRefresh];
}

@end

File: AppLovin MAX Demo App - Swift/AppLovin MAX Demo App - Swift/MAX/MRECs/UI/ALMAXMRecTableViewCell.swift
//
//  ALMAXMRecTableViewCell.swift
//  AppLovin MAX Demo App - Swift
//
//  Created by Alan Cao on 6/27/22.
//  Copyright © 2022 AppLovin. All rights reserved.
//

import UIKit
import AppLovinSDK

class ALMAXMRecTableViewCell: UITableViewCell
{
    private var adView: MAAdView!
    
    override func prepareForReuse()
    {
        super.prepareForReuse()
        
        adView?.removeFromSuperview()
        adView = nil
    }
    
    func configure(with adView: MAAdView)
    {
        self.adView = adView
        self.adView.backgroundColor = .black
        self.adView.translatesAutoresizingMaskIntoConstraints = false
        self.adView.startAutoRefresh()
        
        contentView.addSubview(self.adView)
        NSLayoutConstraint.activate([
            self.adView.widthAnchor.constraint(equalToConstant: 300),
            self.adView.heightAnchor.constraint(equalToConstant: 250),
            self.adView.centerXAnchor.constraint(equalTo: contentView.centerXAnchor),
            self.adView.centerYAnchor.constraint(equalTo: contentView.centerYAnchor),
            self.adView.bottomAnchor.constraint(lessThanOrEqualTo: contentView.bottomAnchor)
        ])
    }
    
    func stopAutoRefresh()
    {
        adView?.setExtraParameterForKey("allow_pause_auto_refresh_immediately", value: "true")
        adView?.stopAutoRefresh()
    }
}
